tests/ui/higher-ranked/subtype/structually-relate-aliases.rs

```diff
@@ -0,0 +1,15 @@
+trait ToUnit<'a> {
```

Comment:
please move this test inte `higher-ranked`, it's not subtyping

please also add "// regression test for issue #121649." to file

r=me after that

Reply:
Done. Thanks for the review!